napari/layers/labels/labels.py

```diff
@@ -248,10 +255,12 @@ def __init__(
         self._contiguous = True
         self._brush_size = 10
 
-        self._selected_label = 1
-        self._selected_color = self.get_color(self._selected_label)
+        self._active_label = 1
+        self._active_color = self.get_color(self._active_label)
         self.color = color
 
+        self._selected_labels = set()
```

Comment:
I was sort of expecting somewhere here to see our [Selection](https://github.com/napari/napari/blob/1c6d8718e0f37e392eb03d59bda79ae7d4ed1274/napari/utils/events/containers/_selection.py#L13) object, which is an evented set, be used. I think that would then give you a lot of nice functionality "for free" - thoughts @tlambert03?

Also cc @andy-sweet who might be interested in this too

Reply:
Ha! yes that is a very good idea @sofroniewn! Thanks for the pointer.

Reply:
Would it work to simply subclass `Selectable`?

Reply:
The way `Selection` works is a bit incompatible with what we want here. It has `active` (a way to select a single element) and `_current` (more similar to what we called `active` here). However, `_current` is considered private *and* it is not automatically updated to always be in sync with `selection`. The docstring says:
```
An item can be the current item and selected at the same time. Qt views
    will ensure that there is always a current item as keyboard navigation,
    for example, requires a current item.
```
I'm not sure what "Qt views" are nor if it's relevant in this case, but I see none of this behaviour in the code.

It makes sense to me that `_current` should be always be one of the currently selected elements (a random one if not explicitly set).

Reply:
Hmm, I'd probably prefer to avoid subclassing at the layer level and use a composition pattern instead of inheritance so you have `layer.selection` be the set and `layer.selection.active` be the singleton. We could also make `_current` public too if we have a need here but maybe @tlambert03 has ideas? If we like this we should probably do this for all our layers with selectable objects - points, shapes etc